### PR TITLE
add kafka library

### PR DESCRIPTION
**Pull request by evonck**
Add kafka library.
The godoc link should work as soon as the PR is merged

### CODE REVIEW

**Comment by spierre** [comment]
tests pass for me

File: kafka/consumer.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+// RegisterEvent register the event into the map
+func (objectConsumer *Consumer) RegisterEvent(eventName string, object *ProcessEvent) {
+	log.Info("Register event ", eventName, " for ", objectConsumer.Topic)
+	(*objectConsumer.eventMap())[eventName] = object
+}
+
+// eventMap event map of the webhook
+func (objectConsumer *Consumer) eventMap() *eventMap {
+	if objectConsumer.eMap == nil {
+		log.Debug("Alloc internal event map for consumer: ", objectConsumer.Topic)
```
Comment (UrielCh) [inline kafka/consumer.go]: it's time you grow up

use Once.Do(func(){}) from sync.Once

File: kafka/producer.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+func (k *Producer) InitKafka(topics []string) {
+	for _, topic := range topics {
+		_, _, err := k.producer.SendMessage(&sarama.ProducerMessage{
+			Topic: topic,
+			Value: sarama.StringEncoder("init message"),
+		})
+		if err != nil {
+			log.Warnf("Failed to send your data %s", err)
+			panic("error during init")
+			return
```
Comment (UrielCh) [inline kafka/producer.go]: is a return is necessary after a panic ?

File: kafka/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+Library kafka containing the basic function used by meastor api micro service to communicate with [kafka](http://kafka.apache.org/).
+
+
+## Dependency
+-  [wvanbergen/kafka](https://github.com/wvanbergen/kafka) 
+-  [Shopify/sarama](https://github.com/Shopify/sarama) 
+
+## Run The Test
+In order to run the test of this library you need to have an kafka server running.
+You can find a docker image of kafka [here](https://hub.docker.com/r/spotify/kafka/)
```
Comment (UrielCh) [inline kafka/README.md]: add comands line to start docker kafka server present in https://hub.docker.com/r/spotify/kafka/

File: kafka/producer.go (hunk 2 above)
Comment (evonck) [inline kafka/producer.go]: I had a case of recovery after panic

File: tools/utils_test.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	convey.Convey("test2 should be in array", t, func() {
+		convey.So(IsInArray("test2", testArray), convey.ShouldEqual, true)
+	})
+	convey.Convey("test should be in array", t, func() {
+		convey.So(IsInArray("test", testArray), convey.ShouldEqual, true)
+	})
+	convey.Convey("test4 should not be in array", t, func() {
+		convey.So(IsInArray("test4", testArray), convey.ShouldEqual, false)
+	})
+
```
Comment (UrielCh) [inline tools/utils_test.go]: big bad empty line

File: kafka/consumer.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+// RegisterEvent register the event into the map
+func (objectConsumer *Consumer) RegisterEvent(eventName string, object *ProcessEvent) {
+	log.Info("Register event", eventName, "for", objectConsumer.Topic)
+	(*objectConsumer.eventMap())[eventName] = object
+}
+
+// eventMap event map of the webhook
+func (objectConsumer *Consumer) eventMap() *eventMap {
+	once.Do(func() {
+		if objectConsumer.eMap == nil {
```
Comment (UrielCh) [inline kafka/consumer.go]: bad place for this if

**Pull request merged**